modules/apps/static/portal-configuration/portal-configuration-metatype/src/main/java/com/liferay/portal/configuration/metatype/util/ConfigurationScopedPidUtil.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.configuration.metatype.util;
+
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * @author Drew Brokke
+ */
+public class ConfigurationScopedPidUtil {
+
+	public static final String PID_SCOPE_SEPARATOR_COMPANY = "__COMPANY__";
+
+	public static final String PID_SCOPE_SEPARATOR_GROUP = "__GROUP__";
+
+	public static final String PID_SCOPE_SEPARATOR_PORTLET_INSTANCE =
+		"__PORTLET_INSTANCE__";
+
+	public static String buildConfigurationScopedPid(
+		String basePid, ExtendedObjectClassDefinition.Scope scope,
+		String scopePrimKey) {
+
+		Objects.requireNonNull(
+			basePid,
+			"The basePid must not be null. A scoped PID must correspond to a " +
+				"configuration PID");
+
+		if ((scope == null) ||
+			scope.equals(ExtendedObjectClassDefinition.Scope.SYSTEM)) {
+
+			return basePid;
+		}
+
+		Objects.requireNonNull(
+			scopePrimKey,
+			"The scopePrimKey must not be null. A scoped PID must correspond " +
+				"to a primary key for its scope.");
+
+		StringBundler sb = new StringBundler(3);
+
+		sb.append(basePid);
+		sb.append(getScopeSeparatorString(scope));
+		sb.append(scopePrimKey);
+
+		return sb.toString();
+	}
+
+	public static String getBasePid(String scopedPid) {
+		if (Validator.isNull(scopedPid)) {
+			return null;
+		}
+
+		if (scopedPid.contains(PID_SCOPE_SEPARATOR_COMPANY)) {
```

Comment:
I think it would be better for the scope enum to have a "separator" field, or just use the automatic string as a separator. That way, if at any time we need to add a new scope, it's only necessary to add a new value to the enum. Otherwise, it would be necessary to look for all the cases where we do this chain of if statements.

Makes sense?

Reply:
Yeah that makes sense.  I hadn't even considered adding anything to the enum but that definitely makes it a lot easier :)